### ISSUE TITLE
CertManager certificate should include both hostnames and domain

### ISSUE THREAD

**Issue by daniilbargman** [Mailu/helm-charts#136]
Hello,

Thank you so much for putting this together!

Building on your example domain name "example.com", the hostname of my email server would be "mail.example.com", whereas all email addresses are created with "...@example.com". When I tried to verify my email headers, the DKIM check kept failing with the following message:

> dkim=fail header.i=@example.com header.s=dkim... 

The default behaviour of the `CertManager` template in the helm chart is to only include the list of hostnames in the certificate. After I added the domain name (which would be "example.com" in the above case) to the list of certificate `dnsNames`, the DKIM check passed.

Unless I am missing something, maybe it would make sense to make this behaviour the default?

Thanks
Daniil

**Comment by micw** [comment]
I'm pretty sure that you missed something but I cannot see, what it is. Adding the mail domains to the cert does not make any sense. Usually, only the MX records points to the MTA, so mailu would have no chance to validate these certs. Also, for DKIM other keys are used and the Cert does not play any role. No idea why it got fixed at yours after adding the domain to the cert but it's definitely not what you want/need.

**Comment by daniilbargman** [comment]
Thank you for the reply. There is one other possibility, namely, that there was some misconfiguration with Gmail servers when I was running my original DKIM tests. And it could have been fixed by the time I updated the certificate, which happened a couple of weeks later. It also seemed strange to me that what I did would solve the problem, because DKIM records are generated independently from the server certificate. I thought maybe there was something peculiar in the Mail source code which made this happen, which is why I opened this thread
 But based on your reply it was probably just a coincidence. I will remove the domain from my DNS records and try again just to be sure, but in any case let me close this one because what you are saying makes total sense. And if I get the same problem again, I will investigate elsewhere.

Thanks again and all the best!
